src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -73,9 +73,13 @@ sealed class NumericRange[T](
 
   // See comment in Range for why this must be lazy.
   override lazy val length: Int = NumericRange.count(start, end, step, isInclusive)
-  override def isEmpty = length == 0
+  override def isEmpty = (
+    (start > end && num.gt(step, num.zero))
+      || (start < end && num.lt(step, num.zero))
+      || (start == end && !isInclusive)
```

Comment:
Should we use `num.gt` / `num.lt`/ `num.equiv` for `start` and `end`? In particular, `==` is maybe incorrect as it doesn't consult the ordering.

Reply:
Update to use `num.gt`/`num.lt`/`num.equiv`.  Not sure why I used it for zero comparisons, but not `start`/`end`